src/enemy/octorokBoss.c

```diff
@@ -0,0 +1,1568 @@
+    }
+}
+
+void OctorokBoss_Hit_SubAction2(Entity* this) {
+    if (GET_HELPER(this)->tailObjects[0]->field_0x7c.BYTES.byte1 != 0x80) {
+        GET_HELPER(this)->tailObjects[0]->field_0x7c.BYTES.byte1++;
+    } else {
+        this->subAction = 3;
+        GET_TIMER(this) = 0x96;
+        gRoomControls.cameraTarget = &gPlayerEntity;
+    }
+}
+
+void OctorokBoss_Hit_SubAction3(Entity* this) {
+    if (GET_TIMER(this)-- == 0) {
+        this->health = OctorokBoss_HealthPerPhase[GET_BOSS_PHASE(this)];
+        COLLISION_ON(this);
+        this->action = ACTION1;
+        this->subAction = ACTION1_AIMTOWARDSPLAYER;
+        GET_NEXT_ATTACK_INDEX(this) = 0;
+        OctorokBoss_SetWaitTurnsForNextAttack(this);
+    }
+}
+
+void OctorokBoss_Hit_SubAction4(Entity* this) {
+    Entity* object;
+    this->subAction = 5;
+    object = CreateObjectWithParent(this, OCTOROK_BOSS_OBJECT, 9, 0);
+    if (object != NULL) {
+        gRoomControls.cameraTarget = object;
+    }
+}
+
+void OctorokBoss_Hit_SubAction5(Entity* this) {
+    this->subAction = 6;
+    GET_TIMER(this) = 0x78;
+    this->field_0x80.HALF.LO = 0;
+    this->field_0x82.HALF.LO = 0;
+}
+
+// Wildly rotating with explosion fx
+void OctorokBoss_Hit_SubAction6(Entity* this) {
+    s16 tmp;
+
+    GET_ANGLE_HI(this) -= 4;
+    this->field_0x80.HALF.HI += 8;
+    if ((this->field_0x80.HALF.LO & 0x80) != 0) {
+        this->field_0x80.HALF.LO -= (Random() & 3);
+        if ((this->field_0x80.HALF.LO & 0x80) == 0) {
+            this->field_0x80.HALF.LO = ((this->field_0x80.HALF.LO & 0x80) << 0x18) >> 0x18;
+        }
+    } else {
+        this->field_0x80.HALF.LO += (Random() & 3);
+        if ((this->field_0x80.HALF.LO) >= 0x19) {
+            this->field_0x80.HALF.LO |= 0x80;
+        }
+    }
+    tmp = FixedMul(gSineTable[this->field_0x80.HALF.HI], (this->field_0x80.HALF.LO & 0x7f) << 8);
+    tmp = FixedDiv(tmp, 0x100);
+    this->spriteOffsetX = tmp >> 8;
+    tmp = FixedMul(gSineTable[this->field_0x80.HALF.HI + 0x40], (this->field_0x80.HALF.LO & 0x7f) * 0x100);
+    tmp = FixedDiv(tmp, 0x100);
+    this->spriteOffsetY = -((tmp << 0x10) >> 8) >> 0x10;
+    if (GET_TIMER(this) == 0) {
+        if ((gScreenTransition.frameCount & 0xfU) == 0) {
+            // Explosion in the center
+            CreateFx(this, 0x48, 0);
+            // Explosion at the front right leg
+            CreateFx(GET_HELPER(this)->legObjects[0], 0x48, 0);
+        }
+        if (++this->field_0x82.HALF.LO == 0x79) {
+            GET_HELPER(this)->mouthObject->health = 1;
+            SoundReq(SFX_BOSS_DIE);
+            // Kill this boss
+            sub_0804A7D4(this);
+        }
+    } else {
+        GET_TIMER(this)--;
+    }
+}
+
+void OctorokBoss_OnTick(Entity* this) {
+    OctorokBoss_Actions[this->action](this);
+    this->spriteRendering.b3 = 3;
+}
+
+ASM_FUNC("asm/non_matching/octorokBoss/OctorokBoss_Init.inc", void OctorokBoss_Init(Entity* this))
+/*{
```

Comment:
Does this compile? It would be better to turn this into a NONMATCH macro so the code can be maintained easier

Reply:
It does compile, but it might not yet be equivalent, which is why I did not put them in NONMATCHING (as that also would count towards the non matching percentage.
Maybe it's better to just keep those beginnings of the decompilation for functions locally until I want to try them again?

Reply:
I think it's fine to use the macro for non equivalent. It's better that we have a starting place for the next person to pick up.

Reply:
I'd expect it to be equivalent if you use the nonmatch macro - the rom should still work if you compile it non-matching.

Reply:
That's fair.. I can't say that's how I've been treating them though, but we will go with it because that's the consensus. With that said I'll open an issue about guaranteeing functional equivalence for non matches.